dace/transformation/interstate/move_loop_into_map.py

```diff
@@ -48,39 +51,57 @@ def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
         if body != body_end:
             return False
 
+        # Body must have only a single connected component
+        # NOTE: This is a strict check that can be potentially relaxed.
+        # If only one connected component includes a Map and the others do not create RW dependencies, then we could
+        # proceed with the transformation. However, that would be a case of an SDFG with redundant computation/copying,
+        # which is unlikely after simplification transformations. Alternatively, we could try to apply the
+        # transformation to each component separately, but this would require a lot more checks.
+        if len(list(nx.weakly_connected_components(body._nx))) > 1:
```

Comment:
```suggestion
        if not permissive and len(list(nx.weakly_connected_components(body._nx))) > 1:
```

Reply:
No, this check is too important to be left to the "permissive" flag. Otherwise, I would have used that in the first place.